### PR TITLE
Merge for v0.9.0

### PR DESCRIPTION
**Pull request by stevenguh**
This release has been in the works for a while, and I think it's about time to merge to master then release it soon after. The changes are documented in CHANGELOG.

Once merge, we should able to close out
#41 
#27 
#13 
#12

@MarcoIeni Would love to have you look at the CHANGELOG and the try it out?

### CODE REVIEW

**Comment by MarcoIeni** [comment]
I will try to find the time to do it in the next couple of days :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,53 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+        "command": "whichkey.toggleZenMode",
+        "when": "transientVisible"
+    }
+    ```
+
+  - No default shortcut is bound at the moment
+  - This is implemented to show more content of the buffer instead being block by the QuickPick menu
+- Add repeater to record and repeat previously executed command ([#27](https://github.com/VSpaceCode/vscode-which-key/issues/27))
+  - Command `whichkey.repeatRecent` will show a which key menu to select all previous command executed in whichkey and bound to `<spc> r .` by default.
+  - Command `whichKey.repeatMostRecent` will execute the recently command executed in whichkey.
```
Comment (MarcoIeni) [inline CHANGELOG.md:43]:
```suggestion
  - Command `whichkey.repeatMostRecent` will execute the recently command executed in whichkey.
```

**Comment by MarcoIeni** [comment]
icon support works fine 
image
---
describe bindings works fine, even though I don't like the name. I was expecting to get an "explanation" of the item when I pressed enter, instead it was executed.
Maybe something like "search binding by name"?
---
I didn't undestand how to try zen mode. Can you add some steps in the changelog?
---
repeats works!
---

**Comment by The-Compiler** [comment]
FWIW icon support seems to break for transient bindings (I think), see https://github.com/VSpaceCode/vscode-which-key/issues/41#issuecomment-856311225

**Comment by stevenguh** [comment]
Thanks for helping out @MarcoIeni.

> describe bindings works fine, even though I don't like the name. I was expecting to get an "explanation" of the item when I pressed enter, instead it was executed.
> Maybe something like "search binding by name"?

The describeBindings name was the expanded name of [`descBinds`](https://github.com/emacs-helm/helm-descbinds) which can search/describe/execute the command. Since the function is basically emulation of descBinds, not sure if there's a better name for it. I guess "searchBindings" would probably works?

---

> I didn't undestand how to try zen mode. Can you add some steps in the changelog?

Currently, there's no default shortcut from us. A user will have to setup their own shortcut for now. For example:
```json
{
	"key": "ctrl+z",
	"command": "whichkey.toggleZenMode",
	"when": "transientVisible"
}
```
Will add a snippet to changelog later this week.

---

> ms to break for transient bindings (I think), see [#41 (comment)](https://github.com/VSpaceCode/vscode-which-key/issues/41#issuecomment-856311225)

Replied in the issue, and will implement/fix it before this version is release.

**Comment by MarcoIeni** [comment]
> I guess "searchBindings" would probably works?
Yes, I think it is mot explanatory

**Comment by stevenguh** [comment]
Forced pushed with following major changes

1. Changed the command name describeBindings to searchBindings
2. Added an example shortcuts for enabling zen mode in transient mode
3. Implemented icons for transient
4. Update README with the suggestion, and add period at the end of all bullet point to be consistent.

**Comment by MarcoIeni** [comment]
Great! I think that if The-Compiler tried it we can merge it :)

**Comment by stevenguh** [comment]
Merged! Thank you everyone. Will release the new version over the weekend if there's no other pressing issue.

**Comment by The-Compiler** [comment]
One thing I noticed is that the extensions are still marked `"preview": true` in `package.json` and thus [in the marketplace](https://marketplace.visualstudio.com/items?itemName=VSpaceCode.vspacecode):

image

Maybe that should be lifted? Or is that something you'd rather do for a v1.0.0 release?

**Comment by stevenguh** [comment]
Yes and I don't think we are at the point we can call v1.0 yet. We are moving there for sure, I think once we tackle the layer problem, we should be in a pretty good spot.  Preview is to at least signal API can changed ;)

**Pull request merged**